### DISCUSSION TITLE
Using parse_obj to convert between BaseModels

### DISCUSSION THREAD

**Discussion by dcheno** [pydantic/pydantic#4948] [Question] [answered]
Hi,

I've been looking for a way to convert between two pydantic models (where the source is a superset of the fields of the target). I noticed that this works:

```python3
from pydantic import BaseModel

class Person(BaseModel):
  name: str
  age: int

class NamelessPerson(BaseModel):
  age: int

mike = Person(name="Mike Jones", age=41)
nameless_mike = NamelessPerson.parse_obj(mike)
```

But I don't see this usage of `parse_obj` anywhere in the docs. I did see it sort of referenced in [this issue](https://github.com/pydantic/pydantic/issues/2474) (via nesting). But the docs on `parse_obj` mention:

>If the object passed is not a dict a ValidationError will be raised.

So I wanted to check, is this an intended part of the API, or should I stay away from it?

If it's an intentional feature, I'd be happy to put up a PR for a doc tweak.

Thanks!

**Comment by anthonycorletti** [comment, marked as answer]
👋 Hey @dcheno. You could also consider [`parse_obj_as`](https://docs.pydantic.dev/usage/models/#parsing-data-into-a-specified-type). For example:

```Python
from pydantic import BaseModel, parse_obj_as


class Person(BaseModel):
  name: str
  age: int

class NamelessPerson(BaseModel):
  age: int

mike = Person(name="Mike Jones", age=41)
nameless_mike = parse_obj_as(NamelessPerson, mike)

assert mike.name == "Mike Jones"
assert nameless_mike.age == mike.age
assert not hasattr(nameless_mike, "name")
```

**Reply by dcheno** [reply]
Thanks!

**Reply by HoverHell** [reply]
What would be the recommended way for converting between models for Pydantic V2? Aside from `NewModel(**old_model_obj.model_dump())`

**Reply by anthonycorletti** [reply]
Good question. I'm not exactly sure. What you're doing looks like an acceptable approach. [`TypeAdapter`](https://docs.pydantic.dev/latest/concepts/type_adapter/) could be a suggested way too.

```python
from pydantic import BaseModel, TypeAdapter


class Person(BaseModel):
  name: str
  age: int

class NamelessPerson(BaseModel):
  age: int

alice = Person(name='Alice', age=42)
nameless_alice = TypeAdapter(NamelessPerson).validate_python(alice.model_dump())

print("alice", alice)
print("nameless_alice", nameless_alice)
```
